### PR TITLE
content with longer name

### PR DESCRIPTION
**Pull request by vijaysawant**
- BZ Close loop #2127470
- Verify that long name CV publish when CV & docker repo both have larger name

### CODE REVIEW

File: tests/foreman/cli/test_container_management.py [hunk 1]
```diff
@@ -322,3 +322,51 @@ def test_positive_container_admin_end_to_end_pull(
...
+        :BZ: 2127470
+
+        :customerscenario: true
+        """
+        pattern_postfix = gen_string('alpha', 50)
+
+        product_name = f'Containers-{pattern_postfix}'
+        repo_name = f'Repo-{pattern_postfix}'
+        lce_name = f'LCE-{pattern_postfix}'
+        cv_name = f'LCE-{pattern_postfix}'
```
Comment (vsedmik) [inline tests/foreman/cli/test_container_management.py:354]:
```suggestion
        cv_name = f'CV-{pattern_postfix}'
```

File: tests/foreman/cli/test_container_management.py [hunk 2]
```diff
@@ -322,3 +322,51 @@ def test_positive_container_admin_end_to_end_pull(
...
+
+        logger.info("Creating product with longer name")
+        product = make_product_wait({'name': product_name, 'organization-id': default_org.id})
```
Comment (vsedmik) [inline tests/foreman/cli/test_container_management.py:357]:
```suggestion
        product = target_sat.cli_factory.make_product_wait({'name': product_name, 'organization-id': default_org.id})
```
We should shift to use `target_sat.cli*` in new tests.

File: tests/foreman/cli/test_container_management.py [hunk 3]
```diff
@@ -322,3 +322,51 @@ def test_positive_container_admin_end_to_end_pull(
...
+        logger.info("Creating docker repo with longer name")
+        repo = _repo(product['id'], name=repo_name, upstream_name=CONTAINER_UPSTREAM_NAME)
+        Repository.synchronize({'id': repo['id']})
```
Comment (vsedmik) [inline tests/foreman/cli/test_container_management.py:360]:
```suggestion
        target_sat.cli.Repository.synchronize({'id': repo['id']})
```
...and lower for the entities make and CV calls too.

**Review by vsedmik — COMMENTED** [review]
Looks like a solid initials :+1: 
I have just put some nitpicks bellow.

In the end you need to assert that pull of that image really works. Something like [this](https://github.com/SatelliteQE/robottelo/blob/c3af9e881b1793dc4c08921794eb31878098e7e2/tests/foreman/api/test_capsulecontent.py#L910), using just one client (podman) should be enough I guess.

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_container_management.py -k test_negative_pull_content_with_longer_name

File: tests/foreman/cli/test_container_management.py [hunk 4]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+        )
+
+        result = container_contenthost.execute(
+            f'docker login -u {settings.server.admin_username}'
+            f' -p {settings.server.admin_password} {target_sat.hostname}'
+        )
+        assert result.status == 0
+        result = container_contenthost.execute(podman_pull_command)
+        assert result.status == 0
+        result = container_contenthost.execute(f'docker logout {target_sat.hostname}')
```
Comment (ogajduse) [inline tests/foreman/cli/test_container_management.py:395]: You are mixing docker and podman. Could you please unify it?

File: tests/foreman/cli/test_container_management.py [hunk 5]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+            {'id': cv['versions'][0]['id'], 'to-lifecycle-environment-id': lce['id']}
+        )
+        # 4. Can pull in docker images
+        podman_pull_command = (
+            f"podman pull --tls-verify=false {target_sat.hostname}/{module_org.label.lower()}-"
+            f"{product.label.lower()}-{repo_name}"
+        )
+
+        result = container_contenthost.execute(
+            f'docker login -u {settings.server.admin_username}'
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:389]:
```suggestion
            f'podman login -u {settings.server.admin_username}'
```

**Review by ogajduse — CHANGES_REQUESTED** [review]
I have done only code-wise review, leaving the test logic review up to other reviewers. 
:+1: for logging

File: tests/foreman/cli/test_container_management.py [hunk 6]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+            f' -p {settings.server.admin_password} {target_sat.hostname}'
+        )
+        assert result.status == 0
+        result = container_contenthost.execute(podman_pull_command)
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:393]:
```suggestion
        assert container_contenthost.execute(podman_pull_command).status == 0
```

File: tests/foreman/cli/test_container_management.py (hunk 4 above)
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:395]:
```suggestion
        assert container_contenthost.execute(f'podman logout {target_sat.hostname}').status == 0 
```

File: tests/foreman/cli/test_container_management.py [hunk 7]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+        assert result.status == 0
+        result = container_contenthost.execute(f'docker logout {target_sat.hostname}')
+        assert result.status == 0
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:396]:
```suggestion
      
```

File: tests/foreman/cli/test_container_management.py [hunk 8]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+        cv = target_sat.cli_factory.make_content_view(
+            {'name': cv_name, 'composite': False, 'organization-id': module_org.id}
+        )
+        ContentView.add_repository({'id': cv['id'], 'repository-id': repo['id']})
+        # 3. Publish & Promote should success
+        ContentView.publish({'id': cv['id']})
+        cv = ContentView.info({'id': cv['id']})
+        ContentView.version_promote(
+            {'id': cv['versions'][0]['id'], 'to-lifecycle-environment-id': lce['id']}
+        )
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:381]: Lets use the sat_object.cv here as well

File: tests/foreman/cli/test_container_management.py [hunk 9]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+        :customerscenario: true
+        """
+        pattern_postfix = gen_string('alpha', 10).lower()
+
+        product_name = f'containers-{pattern_postfix}'
+        repo_name = f'repo-{pattern_postfix}'
+        lce_name = f'lce-{pattern_postfix}'
+        cv_name = f'cv-{pattern_postfix}'
+
+        # 1. Create Product,docker repo,CV and LCE with long name product / repo
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:358]: Non blocking: 
The logger is enough, there is no need to repeat with comments :)

File: tests/foreman/cli/test_container_management.py [hunk 10]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+    def test_negative_pull_content_with_longer_name(
+        self, target_sat, container_contenthost, module_org
+    ):
+        """Verify that long name CV publish when CV & docker repo both have larger name.
+
+        :id: e0ac0be4-f5ff-4a88-bb29-33aa2d874f46
+
+        :steps:
+
+            1. Create Product,docker repo,CV and LCE with long name product / repo
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:335]:
```suggestion
            1. Create a product, docker repo, CV, and LCE with a long name.
```

File: tests/foreman/cli/test_container_management.py [hunk 11]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+            2. Sync the repos
+            3. Add repository to CV, Publish, Promote CV to LCE
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_container_management.py:337]:
```suggestion
            3. Add repository to CV, publish, and then promote CV to LCE
```

File: tests/foreman/cli/test_container_management.py [hunk 12]
```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
...
+        ContentView.publish({'id': cv['id']})
+        cv = ContentView.info({'id': cv['id']})
+        ContentView.version_promote(
+            {'id': cv['versions'][0]['id'], 'to-lifecycle-environment-id': lce['id']}
+        )
+        # 4. Can pull in docker images
+        podman_pull_command = (
+            f"podman pull --tls-verify=false {target_sat.hostname}/{module_org.label.lower()}-"
+            f"{product.label.lower()}-{repo_name}"
+        )
```
Comment (vsedmik) [inline tests/foreman/cli/test_container_management.py:386]: I believe we should test the repo published in the long-named CV and promoted to the long-named LCE at this point (not the basic repo in Library).
```suggestion
        podman_pull_command = (
            f"podman pull --tls-verify=false {target_sat.hostname}/{module_org.label.lower()}-"
            f"{lce_name}-{cv_name}-{product_name}-{repo_name}"
        )
```

**Review by vsedmik — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_container_management.py (hunk 5 above)
Comment (vijaysawant) [reply tests/foreman/cli/test_container_management.py:389]: yea, should not use both services
Thanks for point out

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_container_management.py -k test_negative_pull_content_with_longer_name

**Review by ogajduse — APPROVED** [review]
ACK pending PRT results

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_container_management.py -k test_negative_pull_content_with_longer_name

**Pull request merged**